dask_method/gboeing-demo.ipynb

```diff
@@ -0,0 +1,380 @@
+    "    # we just need one geometry from the left side because\n",
+    "    first_row = grouped_result.iloc[0]\n",
+    "    from_geom = first_row['geometry_from'] # a shapely object\n",
+    "\n",
+    "    # and then convert to a GeoSeries\n",
+    "    to_geoms = gpd.GeoSeries(grouped_result['geometry_to'])\n",
+    "\n",
+    "    # get an array of distances from the GeoSeries comparison\n",
+    "    distances = to_geoms.distance(from_geom)\n",
+    "    return distances.values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Wall time: 43 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "# use dask to calculate distance matrix with geopandas\n",
+    "tall_list = (dd.merge(ddf, gdf, on='tmp_key', suffixes=('_from', '_to'), npartitions=nparts).drop('tmp_key', axis=1))\n",
+    "distances = (tall_list.groupby('id_from').apply(calc_distances, meta=pd.Series()))\n",
+    "computed = distances.compute()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0        0.000000\n",
+       "1    17777.332006\n",
+       "2    35648.613342\n",
+       "3    40575.208147\n",
+       "4    33844.863950\n",
+       "dtype: float64"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# show results\n",
+    "pd.Series(computed[0][:5])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Calculate distance matrix a new way\n",
+    "\n",
+    "Calculate polygons' centroid distances instead of perimeter distances. If we don't need to know the edge-to-edge distance between our polygons, this is superior as it gives us a spatially-representative point for each polygon *and* most importantly allows us to vectorize our distance matrix computation.\n",
+    "\n",
+    "Use a vectorized euclidean distance calculator (euclidean works fine if geometries are projected, as they are in your example data -- if they're *not* projected, use a vectorized great circle distance calculator like I wrote for OSMnx)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "gdf2 = gdf.copy()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Wall time: 148 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "\n",
+    "# convert polygons into xy centroids\n",
+    "centroids = gdf2.centroid\n",
+    "gdf2['x'] = centroids.map(lambda coords: coords.x)\n",
+    "gdf2['y'] = centroids.map(lambda coords: coords.y)\n",
```

Comment:
Possibly but not necessarily. We should discuss when I'm in the office next, including a fuller exposition of the use case to try to consider this challenge outside the box. It would be valuable to vectorize this problem (otherwise, at some point, why use pandas?).